pyccel/codegen/printing/ccode.py

```diff
@@ -1444,6 +1444,9 @@ def _print_Deallocate(self, expr):
         variable_address = self._print(ObjectAddress(expr.variable))
         if expr.variable.is_alias:
             return f'free_pointer({variable_address});\n'
+        if isinstance(expr.variable.dtype, CustomDataType):
+            Pyccel__del = next((method.name for method in expr.variable.cls_base.methods if method.name.endswith('__Pyccel__del__')), None)
+            return(f"{Pyccel__del}({variable_address});\n")
```

Comment:
Why is the IterationError  suppressed? Won't the code fail if `None` is used anyway? Maybe the destructor should be saved explicitly in the ClassDef similarly to what is done in `Module` to make it easy to find?

Reply:
Fixed f31858d2e5402008754c6cf25de3381f8c0bca94.